internal/workflow_testsuite.go

```diff
@@ -281,6 +281,7 @@ func (t *TestActivityEnvironment) SetWorkerStopChannel(c chan struct{}) {
 // SetOnActivityHeartbeatListener sets a listener that will be called when
 // activity heartbeat is called. ActivityInfo is defined in internal package,
 // use public type activity.Info instead.
+// Note: SetOnActivityHeartbeatListener may be called concurrently.
```

Comment:
```suggestion
// Note: The provided listener may be called concurrently.
```

To clarify, we are saying we may call the listener callback concurrently, not related to when `SetOnActivityHeartbeatListener` is called